sda-commons-client-jersey/src/main/java/org/sdase/commons/client/jersey/builder/AbstractBaseClientBuilder.java

```diff
@@ -21,45 +22,32 @@
 abstract class AbstractBaseClientBuilder<T extends AbstractBaseClientBuilder> {
 
   private static final Logger LOG = LoggerFactory.getLogger(AbstractBaseClientBuilder.class);
-
-  /**
-   * The default timeout to wait for data in an established connection. 2 seconds is used as a trade
-   * between "fail fast" and "better return late than no result". The timeout may be changed
-   * according to the use case considering how long a user is willing to wait and how long backend
-   * operations need.
-   */
-  private static final int DEFAULT_READ_TIMEOUT_MS = 2_000;
-
-  /**
-   * The default timeout to wait until a connection is established. 500ms should be suitable for all
-   * communication in the platform. Clients that request information from external services may
-   * extend this timeout if foreign services are usually slow.
-   */
-  private static final int DEFAULT_CONNECTION_TIMEOUT_MS = 500;
-
   /**
    * As default, the client will follow redirects so that redirect status codes are automatically
    * resolved by the client.
    */
   private static final boolean DEFAULT_FOLLOW_REDIRECTS = true;
 
+  private HttpClientConfiguration httpClientConfiguration;
   private JerseyClientBuilder jerseyClientBuilder;
   private final Tracer tracer;
 
   private List<ClientRequestFilter> filters;
 
-  private int connectionTimeoutMillis;
+  private Integer connectionTimeoutMillis;
 
-  private int readTimeoutMillis;
+  private Integer readTimeoutMillis;
 
   private boolean followRedirects;
 
-  AbstractBaseClientBuilder(JerseyClientBuilder jerseyClientBuilder, Tracer tracer) {
+  AbstractBaseClientBuilder(
+      HttpClientConfiguration httpClientConfiguration,
```

Comment:
I think we should document that this configuration must be the same instance that is used by the given `JerseyClientBuilder`. Another option would be to apply the configuration in this class after it has been modified. In the current implementation of the `io.dropwizard.client.JerseyClientBuilder` and the `io.dropwizard.client.HttpClientBuilder` that is used internally by Dropwizard the configuration is read and applied when `build(String)` is called. But we should not rely on the use of mutable configuration objects by the framework. If we do because it is working right now, we should also document that [where we modify the instance](https://github.com/SDA-SE/sda-dropwizard-commons/pull/455/files#diff-d74fe01ce7e7a048fc30666449d1bd56R130-R142) (see below).

As we are talking about code that is only used for backward compatibility, the documentation option seems to be the most simple one in the long term.

We may also document the parameter as deprecated. It won't be needed any more when the deprecated timeout setters are removed.

```suggestion
      @Deprecated HttpClientConfiguration httpClientConfiguration,
```

Reply:
> […] Another option would be to apply the configuration in this class after it has been modified. […]

Yes. It is not necessary anymore that we create the `JerseyClientBuilder` in the `ClientFactory`. I can try to refactor it.

> […] we should not rely on the use of mutable configuration objects by the framework […]

I can also reset it to the previous value in case someone else wants to use the same configuration instance in different builders (could be problematic in terms of thread-safety).

Reply:
I refactored the `AbstractBaseClientBuilder` so it creates the `JerseyClientBuilder`. This removes the need to highlight how the builder should have been configured.